nmigen/build/plat.py

```diff
@@ -395,6 +395,9 @@ def tcl_escape(string):
         def tcl_quote(string):
             return '"' + re.sub(r"([$[\\])", r"\\\1", string) + '"'
 
+        def sdc_quote(string):
+            return "{" + re.sub(r"([${}\\])", r"\\\1", string) + "}"
```

Comment:
Would be nice to have a comment here explaining that this separate method is necessary because SDC does not actually behave like Tcl.

Reply:
Wasn't the whole point of adding this function the fact that *all* Synplify frontends use this quoting?

Reply:
that would be ice40 in addition to the two I already changed, right?
sadly I do not have icecube installed so i cannot test easily. should I change it anyways?

Reply:
Do you have a testcase handy? I can check all of the currently supported toolchains since I have all of them.